lib/fluent/plugin/out_indicative.rb

```diff
@@ -59,7 +60,7 @@ def send_events(events)
 
     payload = {
       apiKey: @api_key,
-      events: events.map do |data|
+      events: events.filter {|data| data[@event_filter_key] != false}.map do |data|
```

Comment:
Is event_filter_key likely to be nil, and does this case need to be handled here?

Reply:
it is - we won't be sending it until I do the rest of the work

Reply:
@vorn there are a few scenarios:

#### 1. `event_filter_key` (the plugin config) is not configured
In this situation the filter will look for a value in the data dict where the key is `nil`, this will return nil and not equate to false, so all data will be let in if there is no filter set (this is the case for the existing tests, which still pass)

#### 2. They key defined by `event_filter_key` does not exist in `data`
If there is no value set for this filter key in the data, the value will equate to nil and not false so the data will be included (this is part of the new tests)

Reply:
I'll tweak it so that it explicitly checks `@event_filter_key` is not null as it's better semantics